test/common/testvm.py

```diff
@@ -69,10 +69,16 @@ def stdchannel_redirected(stdchannel, dest_filename):
             dest_file.close()
 
 class Timeout:
-    def __init__(self, seconds=1, error_message='Timeout'):
+    def __init__(self, seconds=1, error_message='Timeout', machine=None):
         self.seconds = seconds
         self.error_message = error_message
+        self.machine = machine
     def handle_timeout(self, signum, frame):
+        if self.machine:
+            if self.machine.ssh_process:
+                self.machine.ssh_process.terminate()
```

Comment:
Should _kill_ssh_master() which is called by disconnect() always just call terminate()?